src/index.js

```diff
@@ -69,13 +58,42 @@ class Message {
       From: encode(this.networkPrefix, this.from),
       Nonce: this.nonce,
       Value: this.value,
-      Method: this.method,
       GasPrice: this.gasPrice,
       GasLimit: this.gasLimit,
+      Method: this.method,
       Params: this.params
     }
     return message
   }
 }
 
+typeCheck = ({ to, from, nonce, value, method, gasPrice, gasLimit }) => {
+  if (typeof nonce !== 'number') throw new Error('Nonce is not a number')
+
+  if (!to) throw new Error('No to address provided')
+  if (!from) throw new Error('No from address provided')
+
+  if (!nonce) throw new Error('No nonce provided')
+  if (typeof nonce !== 'number') throw new Error('Nonce is not a number')
+  if (nonce > Number.MAX_SAFE_INTEGER)
+    throw new Error('Nonce must be smaller than Number.MAX_SAFE_INTEGER')
+
+  if (!value) throw new Error('No value provided')
+  if (!BigNumber.isBigNumber(value)) throw new Error('Value is not a BigNumber')
+
+  if (!gasPrice) throw new Error('No gas price provided')
+  if (!BigNumber.isBigNumber(gasPrice))
+    throw new Error('Gas price is not a BigNumber')
+
+  if (!gasLimit) throw new Error('No gas limit provided')
+  if (typeof gasLimit !== 'number') throw new Error('Gas Limit is not a number')
+  if (gasLimit > Number.MAX_SAFE_INTEGER)
```

Comment:
One thing we want to do when checking against `Number.MAX_SAFE_INTEGER` - we probably want to convert both of these to `BigNumber` objects to make sure their comparison is sound. Idk what happens if gasLimit is actually bigger than the biggest js number, and _then_ we try to make a `gt` comparison

Reply:
If the number is bigger, it evaluates to false.

Reply:
Yeah, but we can't guarantee that for every number bigger than `Number.MAX_SAFE_INTEGER`, some operate as expected even if they're larger than the biggest num afaik. Learned that bc I faced a 3 day Ujo bug where only _some_ numbers that were bigger than the largest js number acted unexpectedly. Seems like a small change that _could_ cover our arsses

Reply:
You can just do something like:
```js
if (new BigNumber(gasPrice).isGreaterThan(new BigNumber(Number.MAX_SAFE_INTEGER)) {...}
```

Reply:
You were getting bugs evaluating numbers greater than `Number.MAX_SAFE_INTEGER` and getting bugs?

Reply:
I was adding gasEstimates that were not BigNumbers to some other numbers in js. Every time the gasEstimate was bigger than js' biggest number, but only a fraction of the time did adding it return something unexpected

Reply:
https://github.com/hildjj/node-cbor/blob/88cb0c7d1d6ac562ede45502fdde9ac41b3e0e82/lib/encoder.js#L242

Reply:
Maybe we want to do that?

Reply:
Ya that makes sense - making sure its **not** less than or equal to MAX is a safer check!

Reply:
Good catch